scripts/lib/LinkChecker.ts

```diff
@@ -0,0 +1,123 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2023.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import path from "node:path";
+
+const DOCS_ROOT = "./docs";
+const CONTENT_FILE_EXTENSIONS = [".md", ".mdx", ".ipynb"];
+
+export class File {
+  readonly path: string;
+  readonly anchors: string[];
+
+  /**
+   *    path: Path to the file
+   * anchors: Anchors available in the file
+   */
+  constructor(path: string, anchors: string[]) {
+    this.path = path;
+    this.anchors = anchors;
+  }
+}
+
+export class Link {
+  readonly value: string;
+  readonly anchor: string;
+  readonly originFiles: string[];
+  readonly isExternal: boolean;
+
+  /**
+   *  linkString: Link as it appears in source file
+   * originFiles: Paths to source file containing link
+   */
+  constructor(linkString: string, originFiles: string[]) {
+    const splitLink = linkString.split("#", 2);
+    this.value = splitLink[0];
+    this.anchor = splitLink.length > 1 ? `#${splitLink[1]}` : "";
+    this.originFiles = originFiles;
+    this.isExternal = linkString.startsWith("http");
+  }
+
+  /*
+   * Return list of possible paths link could resolve to
+   */
+  possibleFilePaths(originFile: string): string[] {
+    if (this.value === "") {
+      return [originFile];
+    } // link is just anchor
+    if (this.value.startsWith("/images")) {
+      return [path.join("public/", this.value)];
+    }
+
+    let baseFilePath;
+    if (this.value.startsWith("/")) {
+      // Path is relative to DOCS_ROOT
+      baseFilePath = path.join(DOCS_ROOT, this.value);
+    } else {
+      // Path is relative to originFile file
+      baseFilePath = path.join(path.dirname(originFile), this.value);
+    }
+    // Remove trailing '/' from path.join
+    baseFilePath = baseFilePath.replace(/\/$/gm, "");
```

Comment:
You can use a ternary operator

```suggestion
    const relativeToFolder = this.value.startswith("/") ? DOCS_ROOT : path.dirname(originFile);
    // Also remove trailing '/' from path.join
    const baseFilePath = path.join(relativeToFolder, this.value).replace(/\/$/gm, "");
```